Sources/Xcodeproj/generate().swift

```diff
@@ -70,14 +78,43 @@ public func generate(dstdir: String, projectName: String, srcroot: String, modul
     return xcodeprojPath
 }
 
-private func open(_ path: String..., body: ((String) -> Void) throws -> Void) throws {
+import func libc.fgetc
+import var libc.EOF
+
+private func open(_ path: String..., cacheDir: String, body: ((String) -> Void) throws -> Void) throws {
     let path = Path.join(path)
+    // FIXME: Two files can have same name and different path.
+    // Probably can use MD5 hash of path as file name of the cache.
+    let cachePath = Path.join(cacheDir, path.basename)
     let stream = OutputByteStream()
     try body { line in
         stream <<< line
         stream <<< "\n"
     }
-    try fopen(path, mode: .Write) { fp in
-        try fputs(stream.bytes.bytes, fp)
+    // If file and its cache exists, check if new and old
+    // buffer are equal. Only re-write if they're not same. 
+    if path.isFile && cachePath.isFile {
+        var contents = [UInt8]()
+        try fopen(cachePath, mode: .Read) { fp in
+            while true {
+                let c = fgetc(fp)
+                if c == EOF { break }
+                contents += [UInt8(c)]
+            }
+        }
+        // If contents are same then no need to re-write.
+        if contents == stream.bytes.bytes { return }
+    }
+    // Write the cache.
+    try stream.writeToFile(cachePath)
+    // Write the real file.
+    try stream.writeToFile(path)
+}
+
+extension OutputByteStream {
+    func writeToFile(_ path: String) throws {
+        try fopen(path, mode: .Write) { fp in
+            try fputs(self.bytes.bytes, fp)
+        }
```

Comment:
This doesn't belong on OutputByteStream (the intent of which, not currently realized, is that if you wanted to write it to a file you can open an OutputByteStream which writes to the file directly).

ByteString would be a more reasonable place to put this.